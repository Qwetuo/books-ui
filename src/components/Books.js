import React, { Component } from "react";

const API_HOST = process.env.REACT_APP_BOOKS_API || "http://localhost:3000";

class Books extends Component {
  constructor() {
    super();
    this.state = {
      books: []
    };
  }

  async getBooks() {
    const url = `${API_HOST}/books`;
    const response = await fetch(url);
    if (response.ok) {
      const data = await response.json();
      this.setState({
        books: data
      });
    }
  }

  async componentDidMount() {
    await this.getBooks();
  }

  render() {
    return (
      <div>
        <h1>BOOKS</h1>
        {this.state.books.map(book => {
          return <li key={book._id}>{book.title}</li>;
        })}
      </div>
    );
  }
}

export default Books;
