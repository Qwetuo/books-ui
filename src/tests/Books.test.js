import React from 'react';
import Books from '../components/Books';

import ShallowRenderer from 'react-test-renderer/shallow'; // ES6

const renderer = new ShallowRenderer();
renderer.render(<Books />);

it('renders title', () => {
    let result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot()
})

it('renders title and booklist after the component mount', async () => {
    fetch.mockResponseOnce(
        JSON.stringify([
            {
                title: "book3",
                _id: "3"
            },
            {
                title: "book 4",
                _id: "4"
            }
        ])
    )
    const instance = renderer.getMountedInstance();
    await instance.componentDidMount();
    const output = renderer.getRenderOutput();
    expect(output).toMatchSnapshot()
})