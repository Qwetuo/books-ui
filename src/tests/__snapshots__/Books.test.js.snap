// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders title 1`] = `
<div>
  <h1>
    BOOKS
  </h1>
</div>
`;

exports[`renders title and booklist after the component mount 1`] = `
<div>
  <h1>
    BOOKS
  </h1>
  <li>
    book3
  </li>
  <li>
    book 4
  </li>
</div>
`;
